src/core.jl

```diff
@@ -127,7 +132,7 @@ function send_value!(node::Signal, x, timestep)
     # Set the value and do actions
     node.value = x
     for action in node.actions
-        do_action(action, timestep)
+        action_queue[action.recipient.value] = action
```

Comment:
Oh of course, it was getting gc'd. Fixed now, I believe, with an explicit test for the `recipient.value != nothing`